### DISCUSSION TITLE
React JS SocketIo-Client

### DISCUSSION THREAD

**Discussion by Derrick-Tan-D-WEBDEV** [socketio/socket.io#4654] [Q&A] [answered]
Hi Guys,
I tried to subscribe to any topic or either a specific topic.
But I didnt get any message from it.
My backend did received something.

Using React v18 currently.

Appreciated your help🙏.

`    useEffect(() => {
        const socket = io('http://localhost:8128');

        socket.on('connect', () => {
            console.log('connect');
        });
        socket.on('disconnect', () => {
            console.log('disconnect');
        });

        socket.onAny((data) => {
            console.log(data);
        });

        return () => {
            socket.disconnect();
        };
    }, []);
`

**Comment by darrachequesne** [comment, marked as answer]
I think you need to remove the `socket.disconnect()` from the `componentWillUnmount` callback, because `useEffect()` is called twice with React 18 Strict Mode.

Reference: https://beta.reactjs.org/reference/react/StrictMode

**Reply by Derrick-Tan-D-WEBDEV** [reply]
I tried removing the **socket.disconnect()**, but still not working.
Only both of my nodejs server receiving the message.

I remove the <React.StrictMode> as well.

`   
 useEffect(() => {
        const socket = io('http://localhost:8128');

        socket.on('connect', () => {
            console.log('connect');
        });
        socket.on('disconnect', () => {
            console.log('disconnect');
        });

        socket.on('abc', (data) => {
            console.log(data);
        });

        return () => {};
    }, []);
`

**Reply by darrachequesne** [reply]
Please check this guide: https://socket.io/how-to/use-with-react
